src/Quereum.sol

```diff
@@ -1,14 +1,96 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.13;
 
-contract Counter {
-    uint256 public number;
+contract Quereum {
+    struct Question {
+        string question; // the actual question text
+        uint256 expirationTime; // let's use UTC time?
+        uint256 status; // 0 for active 1 for expired 2 for closed
+        address user; // address of the user who posted the question
+        uint256 reward; // reward for answering the question
+        uint256[] responses; // array of response ids
+        bool rewardAllocated; // if the reward has been allocated or not
+    }
+
+    // Mapping of user addresses to their names/balances/questions
+    mapping(address => string) private accounts;
+    mapping(address => uint256) private balances;
+    Question[] private questions;
+    mapping(address => uint256[]) private userQuestions;
+
+    // Register a new user with a chosen name
+    function register(
+        address userAddress,
+        string memory name
+    ) public returns (bool) {
+        require(
+            bytes(accounts[userAddress]).length == 0,
+            "User already registered"
+        );
+        accounts[userAddress] = name;
+        balances[userAddress] = 0; // initialize balance to 0
+        return true;
+    }
+
+    // add balance to the user's account
+    function addBalance(uint256 amount) public returns (bool) {

```

Comment:
This allows a user to add to their balance without actually sending any money over, simply by calling the function with an arbitrary input.

Instead of using amount as a uint256 parameter, can you use msg.value instead?